File: code/naïve.hs
maxPrimeToN n = sieve [2..] 1
  where
    sieve (h:xs) i
      | i == n && isPrime h = h
      | i < n && isPrime h = sieve xs (i + 1)
      | not $ isPrime h = sieve xs i

isPrime n = check 2
  where
    check i
      | i == n = True
      | n `mod` i == 0 = False
      | n `mod` i >= 0 = check $ i + 1

main = do
  putStrLn $ show $ maxPrimeToN 10000

File: code/naïve.py
def get_nth_prime(n):
  j, i = 2, 0
  while True:
    if is_prime(j):
      i += 1
    if i >= n:
      return j
    j += 1


def is_prime(n):
  for i in xrange(2, n):
    if n % i == 0:
      return False
  return True


print get_nth_prime(10000)
